src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -82,10 +78,129 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         const winrt::Windows::Foundation::Size infiniteSize{ std::numeric_limits<float>::infinity(), std::numeric_limits<float>::infinity() };
         Measure(infiniteSize);
         const auto desiredSize = DesiredSize();
-        _EntryAnimationStartPoint(-desiredSize.Height);
+        _OpenAnimationStartPoint(-desiredSize.Height);
         _ContentClipRect({ 0, 0, desiredSize.Width, desiredSize.Height });
     }
 
+    void SearchBoxControl::_PlayOpenAnimation()
+    {
+        if (CloseAnimation().GetCurrentState() == Media::Animation::ClockState::Active)
+        {
+            CloseAnimation().Stop();
+        }
+
+        if (OpenAnimation().GetCurrentState() != Media::Animation::ClockState::Active)
+        {
+            OpenAnimation().Begin();
+        }
+    }
+
+    void SearchBoxControl::_PlayCloseAnimation()
+    {
+        if (OpenAnimation().GetCurrentState() == Media::Animation::ClockState::Active)
+        {
+            OpenAnimation().Stop();
+        }
+
+        if (CloseAnimation().GetCurrentState() != Media::Animation::ClockState::Active)
+        {
+            CloseAnimation().Begin();
+        }
+    }
+
+    // Method Description:
+    // - Sets the search box control to its initial state and calls the initialized callback if it's set.
+    void SearchBoxControl::_Initialize()
+    {
+        _UpdateSizeDependents();
+
+        // Search box is in Visible visibility state by default. This is to make
+        // sure DesiredSize() returns the correct size for the search box.
+        // (DesiredSize() seems to report a size of 0,0 until the control is
+        // visible for the first time, i.e not in Collapsed state).
+        // Here, we set the search box to "Closed" state (and hence Collapsed
+        // visibility) after we've updated the size-dependent properties.
+        VisualStateManager::GoToState(*this, L"Closed", false);
+
+        CloseAnimation().Completed([this](auto&&, auto&&) {
+            CloseAnimation().Stop();
+            VisualStateManager::GoToState(*this, L"Closed", false);
+        });

```

Comment:
If it runs async, does it require us to use `get_weak()` here? (...in case that the callback is called after we've already destroyed ourselves accidentally.)

Reply:
My understanding is that a Control can attach its _own_ handler for its _own_ events without needing `get_weak()`. If a control is destroyed, the handler would never be called because the Dispatcher queue has been destroyed along with the control.

Reply:
_I could be wrong and I wouldn't be surprised_

Reply:
Dispatcher queues are per thread so it can outlive controls. I believe this should work because the `SearchBoxControl` should be the only owner of the `CloseAnimation` and its destruction should kill whatever time XAML is using internally for the animation. I'm just a tiny bit concerned because it's been very buggy in the past and a weak-ref would (hopefully) be more robust. I'm fine with it either way though, because if it does crash it would 100% not be your fault.

Reply:
The handlers are now resilient to an already destructed Control. Thanks.